lib/membrane/video_compositor/handler.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Membrane.VideoCompositor.Handler do
+  @moduledoc """
+  Module defining behaviour of handlers.
+
+  Implementing handler allows to provide custom implementation and
+  react to various events, among others by setting a new scene and/or
+  the inner custom state.
+  """
+
+  alias __MODULE__.CallbackContext
+  alias Membrane.{Pad, StreamFormat, Time}
+  alias Membrane.VideoCompositor.Scene
+
+  @typedoc """
+  Type of user-managed inner state of the handler.
+  """
+  @type state :: any()
+
+  @type video_details :: {pad :: Pad.ref_t(), format :: StreamFormat.t()}
+
+  @typedoc """
+  Type of a valid return value from the callback. By returning this type,
+  the scene will be changed immediate, i.e. at the moment when the event happens.
+  """
+  @type immediate_callback_return :: {scene :: Scene.t(), state :: state()}
+
+  @typedoc """
+  Type of a valid return value from callback allowing to pick start time of a new scene.
+  """
+  @type timed_callback_return :: {{start_ts :: Time.t(), scene :: Scene.t()}, state :: state()}
+
+  @typedoc """
+  Type of a valid return value from callback not changing the current scene.
+  """
+  @type idle_callback_return :: {state :: state()}
+
+  @doc """
+  Callback invoked upon initialization of Video Compositor.
+  """
+  @callback handle_init(ctx :: CallbackContext.Init.t()) :: immediate_callback_return()
```

Comment:
IMO this ctx is quite misleading. Maybe VC options or not ctx at all would be better.

![Screenshot 2023-05-05 at 12 31 09](https://user-images.githubusercontent.com/104033489/236435720-c52ccd48-27fb-446f-ad49-4413950f5cfa.png)

Reply:
`current_scene` should be nil or empty, `input_pads` and `scenes_queues` are just empty lists here, so I think we don't need `ctx` here.